apps/namadillo/src/atoms/fees/services.ts

```diff
@@ -1,14 +1,95 @@
 import {
   DefaultApi,
   GasEstimate,
+  GasLimitTableInnerTxKindEnum,
   GasPriceTableInner,
 } from "@namada/indexer-client";
+import { TxKind } from "types/txKind";
+
+const convertLegacyGasKind = (kind: GasLimitTableInnerTxKindEnum): TxKind => {
+  switch (kind) {
+    case "revealPk":
+      return "RevealPk";
+    case "bond":
+      return "Bond";
+    case "unbond":
+      return "Unbond";
+    case "redelegation":
+      return "Redelegate";
+    case "claimRewards":
+      return "ClaimRewards";
+    case "voteProposal":
+      return "VoteProposal";
+    case "transparentTransfer":
+      return "TransparentTransfer";
+    case "shieldingTransfer":
+      return "ShieldingTransfer";
+    case "shieldedTransfer":
+      return "ShieldedTransfer";
+    case "unshieldingTransfer":
+      return "UnshieldingTransfer";
+    case "ibcMsgTransfer":
+      return "IbcTransfer";
+    case "withdraw":
+      return "Withdraw";
+    default:
+      return "Unknown";
+  }
+};
 
 export const fetchGasEstimate = async (
   api: DefaultApi,
-  params: Parameters<typeof api.apiV1GasEstimateGet>
+  txKinds: TxKind[]
 ): Promise<GasEstimate> => {
-  return (await api.apiV1GasEstimateGet(...params)).data;
+  const counter = (kind: TxKind): number | undefined =>
+    txKinds.filter((i) => i === kind).length || undefined;
+
+  const params = [
+    counter("Bond"),
+    counter("ClaimRewards"),
+    counter("Unbond"),
+    counter("TransparentTransfer"),
+    counter("ShieldedTransfer"),
+    counter("ShieldingTransfer"),
+    counter("UnshieldingTransfer"),
+    counter("VoteProposal"),
+    counter("IbcTransfer"),
+    counter("Withdraw"),
+    counter("RevealPk"),
+    counter("Redelegate"),
+  ];
+
+  // fetch from the estimate endpoint
+  try {
+    return (await api.apiV1GasEstimateGet(...params)).data;
+  } catch (e) {
+    console.error(e);
```

Comment:
Can we place a more descriptive error here? I think it will be important to register when the gas estimate fails. Maybe a message, like `console.error("gas estimate failed", e)` (a better message than that of course)